internal/pkg/model/ext.go

```diff
@@ -25,3 +25,16 @@ func (m *Server) Time() (time.Time, error) {
 func (m *Server) SetTime(t time.Time) {
 	m.Timestamp = t.Format(time.RFC3339Nano)
 }
+
+// GetNewVersion returns Agent version if it is different from ver, otherwise return empty string
+func (m *Agent) GetNewVersion(ver string) string {
+	if m == nil {
```

Comment:
Is this an expected scenario or should we log an error in this case?

Reply:
it's not expected, the function is currently called only after successfully authenticated agent (agent is normally not nil). just additional nil safety check.

Reply:
`CheckDifferentVersion` or `GetDifferentVersion` -> took what you have in the description :-)

Reply:
```CheckDifferentVersion``` sounds good, will update